File: apps/moderator/src/lib/reason.ts
// Formats the moderationReason the moderator writes back to the API so the stored reason carries the
// LLM's verdict + confidence score inline — e.g. "AI review (60% confidence): <the model's reason>".
// Used on both write-back paths (auto-action + human-confirmed removal). The score is what the admin
// and operators see on the removed content, so it travels with the decision rather than being lost.
export function moderationReasonText(input: { verdict: string; confidence: number; reason?: string | null }): string {
  const pct = Math.round((Number.isFinite(input.confidence) ? input.confidence : 0) * 100);
  const head = `AI ${input.verdict} (${pct}% confidence)`;
  const body = (input.reason ?? "").trim();
  return body ? `${head}: ${body}` : head;
}

File: services/scorer/scorer/reason.py
"""Reason formatting — port of ``apps/moderator/src/lib/reason.ts``.

Formats the ``moderationReason`` written back to the API so the stored reason carries the
verdict + confidence inline, e.g. ``"AI review (60% confidence): <reason>"``. The score
travels with the decision rather than being lost.
"""

from __future__ import annotations

import math


def moderation_reason_text(verdict: str, confidence: float, reason: str | None = None) -> str:
    pct = round((confidence if math.isfinite(confidence) else 0.0) * 100)
    head = f"AI {verdict} ({pct}% confidence)"
    body = (reason or "").strip()
    return f"{head}: {body}" if body else head
